pkg/policy/opa/engine.go

```diff
@@ -336,16 +342,23 @@ func (e *Engine) reportPassed(regoData *RegoData) {
 }
 
 // Evaluate Executes compiled OPA queries against the input JSON data
-func (e *Engine) Evaluate(engineInput policy.EngineInput) (policy.EngineOutput, error) {
+func (e *Engine) Evaluate(engineInput policy.EngineInput, filter policy.PreScanFilter) (policy.EngineOutput, error) {
 	// Keep track of how long it takes to evaluate the policies
 	start := time.Now()
 
+	e.regoDataMap = filter.Filter(e.regoDataMap, engineInput)
+
+	// update the rule count
+	e.stats.ruleCount = len(e.regoDataMap)
+
 	// Evaluate the policy against each resource type
 	for k := range e.regoDataMap {
 		// Execute the prepared query.
 		rs, err := e.regoDataMap[k].PreparedQuery.Eval(e.context, rego.EvalInput(engineInput.InputData))
 		if err != nil {
-			zap.S().Debug("failed to run prepared query", zap.Error(err), zap.String("rule", "'"+k+"'"), zap.String("file", e.regoDataMap[k].Metadata.File))
+			// since the eval failed with the policy, we should decrement the total count by 1
+			e.stats.ruleCount--
+			zap.S().Warn("failed to run prepared query", zap.Error(err), zap.String("rule", "'"+k+"'"), zap.String("file", e.regoDataMap[k].Metadata.File))
```

Comment:
Why not keep it Debug? It was wilfully altered to be at Debug level due to https://github.com/accurics/terrascan/issues/709